torch/_subclasses/fake_tensor.py

```diff
@@ -1715,11 +1715,25 @@ def _dispatch_impl(
     ) -> Optional[FakeTensor]:
         flat_args, args_spec = pytree.tree_flatten((args, kwargs))
 
+        # See [subclass inputs] below
+        # NB: If you're seeing a mysterious infinite loop involving fake
+        # tensor, it might be related to this line.  Though I'm not sure
+        # how you'll know to read this comment, as this line won't show up
+        # in the stack trace.
+        has_unrecognized_types = _check_for_subclass(flat_args)
+        if has_unrecognized_types:
```

Comment:
Add a comment that this should be the first thing to happen in FakeTensorMode's torch_dispatch so that people don't end up adding logic above it?